apps/server/src/apps/helpers/assert-required-providers.ts

```diff
@@ -0,0 +1,16 @@
+import { INestApplicationContext, Type } from '@nestjs/common';
+
+type AnyFunction = (...args: any[]) => any;
+
+export const assertRequiredProviders = <TInput = any>(
+	app: INestApplicationContext,
+	tokens: (Type<TInput> | AnyFunction | string | symbol)[]
+): void => {
```

Comment:
```suggestion
export const assertRequiredProviders = (app: INestApplicationContext, tokens: Type[]): void => {
```
AFAIS this typing would be sufficient at the moment. I wouldn't make it more generic if there is currently no need.